fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/BoundFilter.java

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.druid.model.filter;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.util.EnumUtils;
+
+import javax.validation.constraints.Null;
+import java.util.Objects;
+
+/**
+ * Bound filters supports filtering on ranges of dimension values.
+ */
+public class BoundFilter extends DimensionalFilter<BoundFilter> {
+
+    /**
+     * Ordering specified for the range filtering.
+     */
+    public enum Ordering {
+        LEXICOGRAPHIC,
+        ALPHANUMERIC,
+        NUMERIC,
+        STRLEN;
+
+        final String orderingType;
+
+        /**
+         * Constructor.
+         */
+        Ordering() {
+            this.orderingType = EnumUtils.enumJsonName(this);
+        }
+
+        /**
+         * Get the JSON representation of this class.
+         *
+         * @return the JSON representation.
+         */
+        @JsonValue
+        public String toJson() {
+            return orderingType;
+        }
+    }
+
+    private final String lower;
+    private final String upper;
+    private final Boolean lowerStrict;
+    private final Boolean upperStrict;
+    private final Ordering ordering;
+
+    /**
+     * Constructor.
+     *
+     * @param dimension The druid dimension to be filtered
+     * @param lower The lower bound of the dimension value to be filtered (Optional)
+     * @param upper The upper bound of the dimension value to be filtered (Optional)
+     * @param lowerStrict Boolean to enable/ disable strict filtering for lower bounds (Optional)
+     * @param upperStrict Boolean to enable/ disable strict filtering for upper bounds (Optional)
+     * @param ordering The Ordering to be applied for the dimension filtering
+     */
+    public BoundFilter (
```

Comment:
Given that there are optional parameters in this constructor, we should have a separate constructor that accept only required Filter fields and defaults all the optional ones by calling this constructor. That will avoid API callers from doing some "confusing" calls like `BoundFilter(..., false, false, ...)`

Lets put default values for optional parameters in accordance with the default-spec in http://druid.io/docs/latest/querying/filters.html

Reply:
Let's do a constructor that takes only dimension, upper and lower